### DISCUSSION TITLE
Middleware

### DISCUSSION THREAD

**Discussion by bschoche** [microcks/microcks#754] [Q&A] [answered]
Hi,
Is there any functionality that would allow me to add an interceptor to all mock endpoint calls?
I'd like to save all requests to a database before passing them on to the Microks mockes...

**Comment by lbroudoux** [comment]
Hi @bschoche,

As of today, there is no real feature that allows intercepting mock invocation before or after the endpoint has been hit. 
That said, 2 mechanisms could be used or extended for what you'd want to achieve in the future:
* `SCRIPT` dispatcher (see [doc](https://microcks.io/documentation/using/advanced/dispatching/#script-dispatcher)) may be used to access request elements and send them to the location of choice. However, you'd have to code the dispatch logic here.
* `MockInvocationEvent` are published on each mock invocation. As of today, they do not embed elements on incoming requests but I imagine we could enhance them in the future.

WIth no relation to this, we also have started thinking about having a pluggable dispatcher architecture that may generalize the `SCRIPT` dispatcher logic by using something like WASM... It's still very early stage in thinking but may be a good foundations for custom middlewares as well.

What do you think?

**Comment by bschoche** [comment, marked as answer]
Thanks for the response. I guess I'll throw together a reverse-proxy or something.
